crates/libm-test/src/special_case.rs

```diff
@@ -82,15 +68,48 @@ impl MaybeOverride<(f32,)> for SpecialCase {
                 // doesn't seem to happen on x86
                 return XFAIL;
             }
+        }
 
-            if ctx.fname == "lgammaf" || ctx.fname == "lgammaf_r" && input.0 < 0.0 {
-                // loggamma should not be defined for x < 0, yet we both return results
+        if ctx.fname == "sincosf" {
+            let factor_frac_pi_2 = input.0.abs() / f32::consts::FRAC_PI_2;
+            if (factor_frac_pi_2 - factor_frac_pi_2.round()).abs() < 1e-2 {
+                // we have a bad approximation near multiples of pi/2
                 return XFAIL;
             }
         }
```

Comment:
With the implementation now corrected, `sincosf` doesn't need any special allowances in testing
```suggestion
```

Reply:
Awesome!